lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
OK now about the notification message, only the document's id is given.

But you still have the issue of potential duplicates, as you don't exclude from dsl query results the rooms that were in cache.

Maybe following implementation could fix that ?
```javascript
notifyDocumentDelete(request, ids) {
  let cachedRooms
  return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
    .then(result => {
      cachedRooms = result;
      const
        notification = { ... };
      this.kuzzle.notifier.notify(cachedRooms, request, notification);
      return this.kuzzle.services.list.internalCache.remove(this.cacheKeyPrefix + id);
    )}
    .then(() => {
      // (...)
      matchedRooms = this.kuzzle.dsl.test(...);
      matchedRooms = _.difference(matchedRooms, cachedRooms);
      // (...)
    });
```